frontend/src/metabase/query_builder/components/QuestionActions.tsx

```diff
@@ -0,0 +1,229 @@
+      <Tooltip tooltip={bookmarkTooltip}>
+        <BookmarkButton
+          animation={animation}
+          isBookmarked={isBookmarked}
+          onlyIcon
+          icon="bookmark"
+          iconSize={ICON_SIZE}
+          onClick={handleClickBookmark}
+          color={bookmarkButtonColor}
+        />
+      </Tooltip>
+
+      <TippyPopoverWithTrigger
+        key="extra-actions-menu"
+        placement="bottom-end"
+        renderTrigger={({ onClick }) => (
+          <Button
+            onClick={onClick}
+            onlyIcon
+            icon="ellipsis"
+            iconSize={ICON_SIZE}
+          />
+        )}
+        popoverContent={
+          <PopoverContainer>
+            <div>
+              <PLUGIN_MODERATION.QuestionModerationButton
+                question={question}
+                VerifyButton={PopoverButton}
+                verifyButtonProps={buttonProps}
+              />
+            </div>
+            {isDataset && (
+              <div>
+                <PopoverButton
+                  icon="notebook"
+                  onClick={handleEditQuery}
+                  data-testid={ADD_TO_DASH_TESTID}
+                  {...buttonProps}
+                >
+                  {t`Edit query definition`}
+                </PopoverButton>
+              </div>
+            )}
+            {isDataset && (
+              <div>
+                <PopoverButton
+                  icon="label"
+                  onClick={handleEditMetadata}
+                  data-testid={ADD_TO_DASH_TESTID}
+                  {...buttonProps}
+                >
+                  {t`Edit metadata`}
+                  <DatasetMetadataStrengthIndicator dataset={question} />
+                </PopoverButton>
+              </div>
+            )}
+            {!isDataset && (
+              <div>
+                <PopoverButton
+                  icon="dashboard"
+                  onClick={() => onOpenModal(MODAL_TYPES.ADD_TO_DASHBOARD)}
+                  data-testid={ADD_TO_DASH_TESTID}
+                  {...buttonProps}
+                >
+                  {t`Add to dashboard`}
+                </PopoverButton>
+              </div>
+            )}
+            {canWrite && (
+              <div>
+                <PopoverButton
+                  icon="move"
+                  onClick={() => onOpenModal(MODAL_TYPES.MOVE)}
+                  data-testid={MOVE_TESTID}
+                  {...buttonProps}
+                >
+                  {t`Move`}
+                </PopoverButton>
+              </div>
+            )}
+            {!isDataset && canWrite && (
+              <div>
+                <PopoverButton
+                  icon="model"
+                  onClick={() => {
+                    const modal = checkCanBeModel(question)
+                      ? MODAL_TYPES.TURN_INTO_DATASET
+                      : MODAL_TYPES.CAN_NOT_CREATE_MODEL;
+                    onOpenModal(modal);
+                  }}
```

Comment:
Should we extract other click handlers from here the same way you did for "Edit query definition" and "Edit metadata" handlers?